src/pilot/install-director.sh

```diff
@@ -272,69 +301,107 @@ echo "## Updating .bash_profile..."
-##    servers go into maintenance mode in ironic if polled for power state too
-##    aggressively.
-#echo
-#echo "## Patching ironic.conf..."
-#apply_patch "sudo patch -b -s /etc/ironic/ironic.conf ${HOME}/pilot/ironic.patch"
-#echo "## Done."
-#
-## This patches an issue where the  Ironic api service returns http 500 errors
-## https://bugzilla.redhat.com/show_bug.cgi?id=1613995
-#echo
-#echo "## Patching 10-ironic_wsgi.conf"
-#apply_patch "sudo patch -b -s /etc/httpd/conf.d/10-ironic_wsgi.conf ${HOME}/pilot/wsgi.patch"
-#echo "## Done"
+# Install patch on all containers to be patched
+for container in "ironic_pxe_http" "ironic_pxe_tftp" "ironic_conductor" "ironic_api";
+  do
+    run_on_container  "${container}" "dnf install -y patch"
+  done
 
+# Update ironic patches
+for container in "ironic_pxe_http" "ironic_pxe_tftp" "ironic_conductor" "ironic_api" ;
+  do
+    # This hacks in a patch to make conductor wait while completion of configuration job
+    echo
+    echo "## Patching Ironic iDRAC driver job.py on ${container}..."
+    upload_file_to_container "${container}" "${HOME}/pilot/ironic_job.patch" "/tmp/ironic_job.patch"
+    run_on_container "${container}" "patch -b -s /usr/lib/python3.6/site-packages/ironic/drivers/modules/drac/job.py /tmp/ironic_job.patch"
+    run_on_container "${container}" "rm -f /usr/lib/python3.6/site-packages/ironic/drivers/modules/drac/job.pyc"
+    run_on_container "${container}" "rm -f /usr/lib/python3.6/site-packages/ironic/drivers/modules/drac/job.pyo"
+    echo "## Done"
+
+    # This hacks in a patch to create a virtual disk using realtime mode.
+    # Note that this code must be here because we use this code prior to deploying
+    # the director.
+    echo
+    echo "## Patching Ironic iDRAC driver raid.py..."
+    upload_file_to_container "${container}" "${HOME}/pilot/raid.patch" "/tmp/raid.patch"
+    run_on_container "${container}" "patch -b -s /usr/lib/python3.6/site-packages/ironic/drivers/modules/drac/raid.py /tmp/raid.patch"
+    run_on_container "${container}" "rm -f /usr/lib/python3.6/site-packages/ironic/drivers/modules/drac/raid.pyc"
+    run_on_container "${container}" "rm -f /usr/lib/python3.6/site-packages/ironic/drivers/modules/drac/raid.pyo"
+    echo "## Done"
+
+    # This hacks in a patch to define maximum number of retries for the conductor
+    # to wait during any configuration job completion.
+    echo
+    echo "## Patching Ironic iDRAC driver drac.py on ${container} ..."
+    upload_file_to_container "${container}" "${HOME}/pilot/drac.patch" "/tmp/drac.patch"
+    run_on_container "${container}" "patch -b -s /usr/lib/python3.6/site-packages/ironic/conf/drac.py /tmp/drac.patch"
+    run_on_container "${container}" "rm -f /usr/lib/python2.7/site-packages/ironic/conf/drac.pyc"
+    run_on_container "${container}" "rm -f /usr/lib/python2.7/site-packages/ironic/conf/drac.pyo"
+    echo "## Done"
+  done
+
+# Update Drac patches
```

Comment:
Should this be "Update iDRAC patches"?

Reply:
refers to the dracclient libraries ( python-dracclient )